builtin_function.go

```diff
@@ -29,39 +30,22 @@ func (r *Runtime) builtin_Function(args []Value, proto *Object) *Object {
 	return ret
 }
 
-func nativeFuncString(f *nativeFuncObject) Value {
+func (f *nativeFuncObject) source() valueString {
 	return newStringValue(fmt.Sprintf("function %s() { [native code] }", nilSafe(f.getStr("name", nil)).toString()))
 }
 
+func (f *baseJsFuncObject) source() valueString {
+	return newStringValue(f.src)
+}
+
 func (r *Runtime) functionproto_toString(call FunctionCall) Value {
 	obj := r.toObject(call.This)
-repeat:
 	switch f := obj.self.(type) {
-	case *funcObject:
-		return newStringValue(f.src)
-	case *classFuncObject:
-		return newStringValue(f.src)
-	case *methodFuncObject:
-		return newStringValue(f.src)
-	case *arrowFuncObject:
-		return newStringValue(f.src)
-	case *nativeFuncObject:
-		return nativeFuncString(f)
-	case *boundFuncObject:
-		return nativeFuncString(&f.nativeFuncObject)
-	case *wrappedFuncObject:
-		return nativeFuncString(&f.nativeFuncObject)
-	case *lazyObject:
-		obj.self = f.create(obj)
-		goto repeat
+	case funcObjectImpl:
+		return f.source()
```

Comment:
Pulled over this change from upstream, could have probably recreated this by just adding templatedFuncObject as one of the switch cases.

Reply:
Do you mean this was not part of the template commit?

Reply:
Correct, this was part of the giant [async/await](https://github.com/dop251/goja/commit/33bff8fdda616056a9eb9b42853e12e9feafc8f9) update